xmtp_mls/src/groups/mls_sync.rs

```diff
@@ -2047,6 +2049,11 @@ async fn apply_update_group_membership_intent(
     let leaf_nodes_to_remove: Vec<LeafNodeIndex> =
         get_removed_leaf_nodes(openmls_group, &changes_with_kps.removed_installations);
 
+    if leaf_nodes_to_remove.contains(&openmls_group.own_leaf_index()) {
```

Comment:
This is where we prevent creating commits that try to do a self remove (for example, when you have been revoked by another installation but don't know it yet)